src/services/observables/tokens.ts

```diff
@@ -399,59 +407,73 @@ export const poolTokens$ = combineLatest([
       converter: string;
     } | null>(
       partialPoolTokens.map(async (poolToken) => {
-        const pool = pools.find(
+        let pool: APIPool | Pool;
+        const poolExists = pools.find(
           (x) => x.converter_dlt_id === poolToken.converter
         );
 
-        if (pool) {
-          const tkn = tokens.find(
-            (x) => x.address === pool.reserves[0].address
-          );
-          const bnt = tokens.find(
-            (x) => x.address === pool.reserves[1].address
+        if (poolExists) {
+          pool = poolExists;
+        } else {
+          const apiPoolExists = apiData.pools.find(
+            (x) => x.converter_dlt_id === poolToken.converter
           );
+          if (apiPoolExists) {
+            pool = apiPoolExists;
+          } else {
+            return null;
+          }
+        }
+        const tkn = pool.reserves[0];
+        const bnt = pool.reserves[1];
 
-          const amount = shrinkToken(poolToken.balance, pool.decimals);
-          const percent = new BigNumber(amount).div(
-            shrinkToken(poolToken.totalSupply, pool.decimals)
+        const amount = shrinkToken(poolToken.balance, pool.decimals);
+        const percent = new BigNumber(amount).div(
+          shrinkToken(poolToken.totalSupply, pool.decimals)
+        );
+
+        const balances = await multicall(
+          pool.reserves.map((x) =>
+            buildTokenPoolCall(pool.converter_dlt_id, x.address)
+          )
+        );
+        if (balances) {
```

Comment:
will be more readable to check !balances and return null

Reply:
done